kvbc/src/v4blockchain/detail/keys_history.cpp

```diff
@@ -0,0 +1,293 @@
+// Concord
+//
+// Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#include "util/endianness.hpp"
+#include "v4blockchain/detail/keys_history.h"
+#include "v4blockchain/detail/column_families.h"
+#include "v4blockchain/detail/blockchain.h"
+#include "rocksdb/details.h"
+#include "build/kvbc/cmf/categorized_kvbc_msgs.cmf.hpp"
+
+using namespace concord::kvbc;
+namespace concord::kvbc::v4blockchain::detail {
+
+KeysHistory::KeysHistory(
+    const std::shared_ptr<concord::storage::rocksdb::NativeClient>& native_client,
+    const std::optional<std::map<std::string, concord::kvbc::categorization::CATEGORY_TYPE>>& categories)
+    : native_client_{native_client}, category_mapping_(native_client, categories) {
+  if (native_client_->createColumnFamilyIfNotExisting(v4blockchain::detail::KEYS_HISTORY_CF,
+                                                      KHCompactionFilter::getFilter())) {
+    LOG_INFO(V4_BLOCK_LOG, "Created [" << v4blockchain::detail::KEYS_HISTORY_CF << "] column family for keys history");
+  }
+}
+
+void KeysHistory::addBlockKeys(const concord::kvbc::categorization::Updates& updates,
+                               const BlockId block_id,
+                               storage::rocksdb::NativeWriteBatch& write_batch) {
+  LOG_DEBUG(V4_BLOCK_LOG, "Adding keys of block [" << block_id << "] to the keys history CF");
```

Comment:
let's also add a feature flag, that, when disabled returns without updating the keys

Reply:
added enableKeysHistoryCF